cantusdb/turtle/csv2rdf_by_RDFLib.py

```diff
@@ -0,0 +1,71 @@
+from rdflib import Graph, URIRef, Literal
+from rdflib.namespace import RDF
+import csv
+import validators
+import sys
+
+# helper to get the predicate relation dictionary
+def get_ontology_dict () :
+    
+    # a specific config file for the predicates
+    # TODO: maybe change this part to make a more customized way of getting the predicates configurations
+    ontology_f = open("ontology_dict.csv", 'r')
+    ontology_reader = csv.DictReader(ontology_f)
+    ontology_dict = {}
+    
+    # converting the reader dict to the predicate dictionary
+    for row in ontology_reader:
+        ontology_dict[row[list(row.keys())[0]]] = row[list(row.keys())[1]]
```

Comment:
I have no idea what this line is doing. Do you have an example output? Or better yet, just change the format of your file to match what you need instead of trying to fit it all into a.CSV.

Reply:
Hi, here is an example input:
relation,URI
absolute_url,
incipit,http://www.wikidata.org/prop/direct/P1922
genre,https://schema.org/genre
genre_id_url,http://www.wikidata.org/prop/direct/P136
mode_name,https://schema.org/musicalKey
mode_id_url,https://www.wikidata.org/wiki/Q731978
finalis,https://www.wikidata.org/wiki/Q4484726
src_link,https://cantusdatabase.org/sources
cantus_id,https://cantusindex.org/id
indexing_notes,https://www.wikidata.org/wiki/Q263478

Goes to output

{'absolute_url': '', 
'incipit': 'http://www.wikidata.org/prop/direct/P1922', 
'genre': 'https://schema.org/genre', 
'genre_id_url': 'http://www.wikidata.org/prop/direct/P136', 
'mode_name': 'https://schema.org/musicalKey', 
'mode_id_url': 'https://www.wikidata.org/wiki/Q731978', 
'finalis': 'https://www.wikidata.org/wiki/Q4484726', 
'src_link': 'https://cantusdatabase.org/sources', 
'cantus_id': 'https://cantusindex.org/id', 
'indexing_notes': 'https://www.wikidata.org/wiki/Q263478'}

It changes an input CSV file into a dictionary line by line. I think there's a clearer way of doing this, I will figure it out today.

Reply:
I guess I'm suggesting creating a file called "mapping.json", or even better (since it's not very large) just keeping the mapping in your code. Then you actually don't need to convert from a CSV to the dictionary.